solutions/agents/variables.tf

```diff
@@ -0,0 +1,222 @@
+  default     = true
+}
+
+variable "cloud_monitoring_access_key" {
+  type        = string
+  description = "Access key used by the IBM Cloud Monitoring agent to communicate with the instance."
+  sensitive   = true
+  default     = null
+}
+
+variable "cloud_monitoring_secret_name" {
+  type        = string
+  description = "The name of the secret which will store the access key."
+  default     = "sysdig-agent"
+  nullable    = false
+}
+
+variable "cloud_monitoring_instance_region" {
+  type        = string
+  description = "The IBM Cloud Monitoring instance region. Used to construct the ingestion endpoint."
+  default     = "us-south"
+}
+
+variable "cloud_monitoring_endpoint_type" {
+  type        = string
+  description = "Specify the IBM Cloud Monitoring instance endpoint type (public or private) to use. Used to construct the ingestion endpoint."
+  default     = "private"
+  validation {
+    error_message = "The specified endpoint_type can be private or public only."
+    condition     = contains(["private", "public"], var.cloud_monitoring_endpoint_type)
+  }
+}
+
+variable "cloud_monitoring_metrics_filter" {
+  type = list(object({
+    type = string
+    name = string
+  }))
+  description = "To filter custom metrics, specify the Cloud Monitoring metrics to include or to exclude. See https://cloud.ibm.com/docs/monitoring?topic=monitoring-change_kube_agent#change_kube_agent_inc_exc_metrics."
+  default     = [] # [{ type = "exclude", name = "metricA.*" }, { type = "include", name = "metricB.*" }]
+  validation {
+    condition     = length(var.cloud_monitoring_metrics_filter) == 0 || can(regex("^(include|exclude)$", var.cloud_monitoring_metrics_filter[0].type))
+    error_message = "Invalid input for `cloud_monitoring_metrics_filter`. Valid options for 'type' are: `include` and `exclude`. If empty, no metrics are included or excluded."
+  }
+}
+
+variable "cloud_monitoring_agent_tags" {
+  type        = list(string)
+  description = "List of tags to associate to all matrics that the agent collects. NOTE: Use the 'cloud_monitoring_add_cluster_name' variable to add the cluster name as a tag."
+  default     = []
+  nullable    = false
+}
+
+variable "cloud_monitoring_add_cluster_name" {
+  type        = bool
+  description = "If true, configure the cloud monitoring agent to attach a tag containing the cluster name to all metric data."
+  default     = true
+}
+
+variable "cloud_monitoring_agent_name" {
+  description = "Cloud Monitoring agent name. Used for naming all kubernetes and helm resources on the cluster."
+  type        = string
+  default     = "sysdig-agent"
+}
+
+variable "cloud_monitoring_agent_namespace" {
+  type        = string
+  description = "Namespace where to deploy the Cloud Monitoring agent. Default value is 'ibm-observe'."
+  default     = "ibm-observe"
+  nullable    = false
+}
+
+variable "cloud_monitoring_agent_tolerations" {
+  description = "List of tolerations to apply to Cloud Monitoring agent."
```

Comment:
Can you update the description to explain what the default values does. I guess it will match any nod except the master